File: core/src/tables/post.rs
use crate::parser::*;
use crate::{TitleKind, Result};

pub fn parse(table_end_offset: usize, parser: &mut Parser) -> Result<()> {
    let version = parser.read::<Fixed>("Version")?;
    parser.read::<Fixed>("Italic angle")?;
    parser.read::<i16>("Underline position")?;
    parser.read::<i16>("Underline thickness")?;
    parser.read::<u32>("Is fixed pitch")?;
    parser.read::<u32>("Min memory when font is downloaded")?;
    parser.read::<u32>("Max memory when font is downloaded")?;
    parser.read::<u32>("Min memory when font is downloaded as a Type 1")?;
    parser.read::<u32>("Max memory when font is downloaded as a Type 1")?;

    if version.0 != 2.0 {
        return Ok(());
    }

    let num_glyphs = parser.read::<u16>("Number of glyphs")?;
    if num_glyphs != 0 {
        parser.read_array::<u16>("Glyph name indexes", TitleKind::Index, num_glyphs as usize)?;
    }

    while parser.offset() < table_end_offset {
        parser.begin_group("Name");
        let len = parser.read::<u8>("Length")?;
        if len != 0 {
            parser.read_string(len as usize, TitleKind::String, None)?;
        }

        parser.end_group();
    }

    Ok(())
}

File: src/tables/post.cpp
#include "src/parser.h"
#include "tables.h"

void parsePost(const quint32 end, Parser &parser)
{
    const auto version = parser.read<Fixed>("Version");
    parser.read<Fixed>("Italic angle");
    parser.read<Int16>("Underline position");
    parser.read<Int16>("Underline thickness");
    parser.read<UInt32>("Is fixed pitch");
    parser.read<UInt32>("Min memory when font is downloaded");
    parser.read<UInt32>("Max memory when font is downloaded");
    parser.read<UInt32>("Min memory when font is downloaded as a Type 1");
    parser.read<UInt32>("Max memory when font is downloaded as a Type 1");

    if (version != 2.0f) {
        return;
    }

    const auto numGlyphs = parser.read<UInt16>("Number of glyphs");
    if (numGlyphs != 0) {
        parser.beginGroup("Glyph name indexes");

        for (auto i = 0; i < numGlyphs; ++i) {
            parser.read<UInt16>("Index");
        }

        parser.endGroup();
    }

    while (parser.offset() < end) {
        parser.beginGroup("");
        const auto len = parser.read<UInt8>("Length");
        if (len != 0) {
            const auto name = parser.readString(len, "Data");
            parser.endGroup(name);
        } else {
            parser.endGroup();
        }
    }
}
